src/task_queue/nudges/user_event_nudge.py

```diff
@@ -107,37 +107,41 @@ def update_last_notification_dates(email):
 
 
 def is_event_eligible(event, community_id, task=None):
-    now = timezone.now()
-    settings = event.nudge_settings.filter(communities__id=community_id).first()
-
-    # it it doesn't exist, then create, don't save an instance of NudgeSettings
-    if not settings:
-        settings = EventNudgeSetting.objects.create(event=event, **DEFAULT_EVENT_SETTINGS)
-        settings.communities.add(community_id)
-
-    if settings.never:
+    try:
+        now = timezone.now().date()
+        settings = event.nudge_settings.filter(communities__id=community_id).first()
+        
+        # it it doesn't exist, then create, don't save an instance of NudgeSettings
+        if not settings:
+            settings = EventNudgeSetting.objects.create(event=event, **DEFAULT_EVENT_SETTINGS)
+            settings.communities.add(community_id)
+        
+        if settings.never:
+            return False
+        
+        freq = task.frequency if task else None
+        last_last_run = None
+        
+        freq_to_delta = {
+            "EVERY_WEEK": relativedelta(weeks=1),
+            "bi-weekly": relativedelta(weeks=2),
+            "EVERY_MONTH": relativedelta(months=1),
+            "EVERY_DAY": relativedelta(days=1)
+        }
+        
+        if freq:
+            last_last_run = now - freq_to_delta.get(freq, relativedelta(days=0))
+        
+        if settings.when_first_posted and event.published_at and last_last_run < event.published_at.date() <= now:
+            return True
+        elif settings.within_30_days and event.start_date_and_time.date() - now <= timezone.timedelta(days=30):
```

Comment:
this will typically send a nudge about the event every week once within 30 days.  Could change it to:
`elif settings.within_30_days and timezone.timedelta(days=30) - last_last_run < event.start_date_and_time.date() - now <= timezone.timedelta(days=30):`